tools/distpartitioning/dataset_utils.py

```diff
@@ -242,23 +243,19 @@ def get_dataset(input_dir, graph_name, rank, world_size, num_parts, schema_map):
         for etype_name, etype_feature_data in dataset_features.items():
             for feat_name, feat_data in etype_feature_data.items():
                 assert feat_data[constants.STR_FORMAT][constants.STR_NAME] == constants.STR_NUMPY
-                num_chunks = len(feat_data[constants.STR_DATA])
-                read_list = np.array_split(np.arange(num_chunks), num_parts)
                 for local_part_id in range(num_parts):
                     if map_partid_rank(local_part_id, world_size) == rank:
                         efeats = []
                         efeat_tids = []
-                        for idx in read_list[local_part_id]:
-                            feature_fname = feat_data[constants.STR_DATA][idx]
-                            if (os.path.isabs(feature_fname)):
-                                logging.info(f'Loading numpy from {feature_fname}')
-                                efeats.append(torch.from_numpy(np.load(feature_fname)))
-                            else:
-                                numpy_path = os.path.join(input_dir, feature_fname)
-                                logging.info(f'Loading numpy from {numpy_path}')
-                                efeats.append(torch.from_numpy(np.load(numpy_path)))
+                        start, end = edge_tids[etype_name][local_part_id]
```

Comment:
Why not use the same loading method as before?  You have a read_list and each process load part of the files in a round-robin way.

Reply:
read behavior is changed to previous one. and shuffle data is applied.